core/node/metadata_calculator/src/updater.rs

```diff
@@ -184,28 +187,38 @@ impl TreeUpdater {
     async fn step(
         &mut self,
         mut storage: Connection<'_, Core>,
-        next_l1_batch_to_seal: &mut L1BatchNumber,
+        next_l1_batch_to_process: &mut L1BatchNumber,
     ) -> anyhow::Result<()> {
-        let Some(last_sealed_l1_batch) = storage
-            .blocks_dal()
-            .get_sealed_l1_batch_number()
-            .await
-            .context("failed loading sealed L1 batch number")?
-        else {
-            tracing::trace!("No L1 batches to seal: Postgres storage is empty");
-            return Ok(());
+        let last_l1_batch_with_protective_reads = if self.sealed_batches_have_protective_reads {
```

Comment:
if `self.tree.mode()` is "light" then it should use `get_sealed_l1_batch_number` regardless config var